src/lua/api-gateway/redis/redisConnectionProvider.lua

```diff
@@ -101,12 +101,19 @@ function RedisConnectionProvider:connectToRedis(host, port, password, redisTimeo
     redis:set_timeout(redis_timeout)
 
     local ok, err = redis:connect(host, port)
-
     if not ok then
         ngx.log(ngx.ERR, "Failed to connect to Redis instance: " .. host .. ", port: " .. port .. ". Error: ", err)
         return false, nil
     end
 
+    -- Check for existing connection
+    local times, error = redis:get_reused_times()
+
+    if times and times > 0 then
```

Comment:
I think you should check that times is not 0 ->
"If the connection comes from the connection pool, then the return value is always non-zero. "